kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/group/GroupMetadataManager.java

```diff
@@ -673,7 +673,7 @@ private void removeProducerGroup(long producerId,
                 }
             }
             openGroupsForProducer.computeIfAbsent(producerId, (pid) -> new HashSet<>())
```

Comment:
Should we replace the code like this? 
```
openGroupsForProducer.computeIfAbsent(producerId, (pid) -> new HashSet<>())
    .remove(groupId);
if (openGroupsForProducer.get(producerId).isEmpty()) {
    openGroupsForProducer.remove(producerId);
}
```

See https://github.com/apache/kafka/blob/8f5c2342244603cea142a2d9d04ba01aa8e14bc6/core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala#L950-L954

Reply:
+1

Reply:
@ZiyaoWei